resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/k8s/PodBuilderSuite.scala

```diff
@@ -66,6 +66,69 @@ abstract class PodBuilderSuite extends SparkFunSuite {
     assert(pod.container.getVolumeMounts.asScala.exists(_.getName == "so_long_two"))
   }
 
+  test("SPARK-37145: configure a custom test step with base config") {
+    val client = mockKubernetesClient()
+    val sparkConf = baseConf.clone()
+      .set(userFeatureStepsConf.key,
+          "org.apache.spark.deploy.k8s.TestStepWithConf")
+      .set(templateFileConf.key, "template-file.yaml")
+      .set("test-features-key", "test-features-value")
+    val pod = buildPod(sparkConf, client)
+    verifyPod(pod)
+    val metadata = pod.pod.getMetadata
+    assert(metadata.getAnnotations.containsKey("test-user-feature-annotation"))
+  }
+
+  test("SPARK-37145: configure a custom test step with driver or executor config") {
+    val client = mockKubernetesClient()
+    val (featureSteps, annotationKey) = this.getClass.getSimpleName match {
```

Comment:
We already selecting `userFeatureStepsConf` based on the concrete implementation (although I would prefer helper functions in a base suite which would be used in the concrete implementation to create specific tests with the right parameterization, but at least let's follow the existing pattern).

So let's have 
```scala 
protected def userFeatureStepWithExpectedAnnotation: (class[_], String)
```

The `TestStepWithDrvConf` and `TestStepWithExecConf ` should be moved into the specific suites (where the annotation content can be accessed by a `val`and reused in these two custom steps and in the `userFeatureStepWithExpectedAnnotation` function).

Reply:
Looks like we can't load a inner class, so I just keep `TestStepWithExecConf` with same level with `KubernetesExecutorBuilderSuite`.

```scala
org.apache.spark.scheduler.cluster.k8s.KubernetesExecutorBuilderSuite.TestStepWithExecConf
java.lang.ClassNotFoundException: org.apache.spark.scheduler.cluster.k8s.KubernetesExecutorBuilderSuite.TestStepWithExecConf
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
```